### PR TITLE
update go to 1.17 and update csv version to 4.11.0

### PR DESCRIPTION
**Pull request by iamniting**
- Update golang to 1.17 and its deps
- Update all actions to make them compatible with golang 1.17
- Update CSV version
- Use go install instead of depricated go get

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: .github/workflows/codespell.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ jobs:
       - name: codespell
         uses: codespell-project/actions-codespell@master
         with:
+          exclude_file: go.sum
```
Comment (umangachapagain) [inline .github/workflows/codespell.yaml:20]: Why do we need to exclude it?

**Comment by umangachapagain** [comment]
@iamniting There are multiple changes in this PR (not just Go 1.17 change).
I suggest updating PR title and description to make it clearer.

File: .github/workflows/codespell.yaml (hunk 1 above)
Comment (iamniting) [reply .github/workflows/codespell.yaml:20]: It was failing here https://github.com/red-hat-storage/odf-operator/actions/runs/2050857294

**Comment by iamniting** [comment]
> @iamniting There are multiple changes in this PR (not just Go 1.17 change). I suggest updating PR title and description to make it clearer.

done

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**